jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/cm/processor/BaseFileProcessor.java

```diff
@@ -0,0 +1,179 @@
+        fps.setConfiguration(config);
+
+        String fName = (String) getPropertyOrNull(config, cName + ".filename");
+        String tempFileName = "";
+
+        if (StringUtils.isBlank(fName)) {
+            tempFileName = filename;
+            log.info("processor configured to load default file: {}", tempFileName);
+        } else {
+            String baseDir = (String) getPropertyOrNull(config, "path.base");
+            if (baseDir != null) {
+                tempFileName = baseDir;
+                if (!baseDir.endsWith(File.separator)) {
+                    tempFileName += File.separator;
+                }
+            }
+            tempFileName += fName;
+
+            log.info("processor configured to load override file: {}", tempFileName);
+        }
+
+        fps.setFilename(tempFileName);
+
+        int numCols = getColumns();
+
+        if (config.containsKey(cName + ".columns")) {
+            String cols = (String) config.get(cName + ".columns");
+
+            try {
+                numCols = Integer.parseInt(cols);
+            } catch (NumberFormatException nfe) {
+                log.error("improper number format specified in ProcessorState configuration for property {} .columns", cName);
+            }
+        }
+
+        fps.setColumns(numCols);
+
+        boolean headerRow = headerRowPresent;
+        if (config.containsKey(cName + ".headerRowPresent")) {
+            String hRows = (String) config.get(cName + ".headerRowPresent");
+            headerRow = Boolean.parseBoolean(hRows);
+        }
+
+        fps.setHeaderRowPresent(headerRow);
+
+        return fps;
+    }
+
+    protected static final Object getPropertyOrNull(final Map<String, Object> config, final String key) {
+        if (config != null) {
+            return config.get(key);
+        }
+        return null;
+    }
+
+    public void process(ProcessorState ps) throws Exception {
+
+        if (!FileProcessorState.class.isAssignableFrom(ps.getClass())) {
+            log.error("could not proceed with ProcessorState of type {}; it is not of type FileProcessorState", ps.getClass().getCanonicalName());
+            throw new Exception("process(...) called with a ProcessorState which is not a FileProcessorState");
+        }
+
+        final FileProcessorState state = (FileProcessorState) ps;
+
+        state.setStartDate(new java.util.Date());
+
+        log.info("{} started {}", getProcessorTitle(), state.getStartDate());
+
+        File dataFile = null;
+        String filename = state.getFilename();
+
+        dataFile = new File(filename);
+
+        if (!dataFile.exists()) {
+            log.error("{} not found", filename);
+            state.appendError(filename + " not found");
+            log.error("{} ended {}", getProcessorTitle(), state.getEndDate());
+            return;
+        }
+
+        BufferedReader fr = new BufferedReader(new FileReader(dataFile));
+
+        try {
```

Comment:
You can use the new Java try with resources here, then you don't have to worry about closing the `BufferedReader`.